src/main/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/StatusAndTiming.java

```diff
@@ -301,9 +304,26 @@ public static GenericStatus computeChunkStatus2(@Nonnull WorkflowRun run,
                 return GenericStatus.FAILURE;
             }
         }
+        WarningAction warning = findWarningBetween(firstNode, lastNode);
+        if (warning != null) {
+            return GenericStatus.UNSTABLE;
+        }
 
         // Previous chunk before end. If flow continued beyond this, it didn't fail.
-        return (run.getResult() == Result.UNSTABLE) ? GenericStatus.UNSTABLE : GenericStatus.SUCCESS;
+        return GenericStatus.SUCCESS;
+    }
+
+    private static @CheckForNull WarningAction findWarningBetween(@Nonnull FlowNode start, @Nonnull FlowNode end) {
+        // TODO: Cache the result?
```

Comment:
Loading a single Pipeline in Blue Ocean will end up calling this method 2-3 times for every stage/branch, and if they are successful this method will need to scan through the whole block each time. It is easy enough to add a `LoadingCache` keyed by the run display name and the start and end nodes (or just the start and end nodes via reference equality?) that stores entries only for a few minutes. I guess we'd have to do some testing with jenkins-scalability-lab to have any idea of the performance with and without a cache.